backend/infrahub/core/query/resource_manager.py

```diff
@@ -102,6 +103,149 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> No
         self.return_labels = ["pool", "rel", "address"]
 
 
+class NumberPoolGetAllocated(Query):
+    name: str = "numberpool_get_allocated"
+
+    def __init__(
+        self,
+        pool: CoreNumberPool,
+        **kwargs: dict[str, Any],
+    ) -> None:
+        self.pool = pool
+
+        super().__init__(**kwargs)  # type: ignore[arg-type]
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:
+        self.params["pool_id"] = self.pool.get_id()
+        self.params["node_attribute"] = self.pool.node_attribute.value
+        self.params["start_range"] = self.pool.start_range.value
+        self.params["end_range"] = self.pool.end_range.value
+
+        self.params["time_at"] = self.at.to_string()
+
+        def rel_filter(rel_name: str) -> str:
+            return f"{rel_name}.from <= $time_at AND ({rel_name}.to IS NULL OR {rel_name}.to >= $time_at)"
+
+        query = f"""
+        MATCH (n:%(node)s)-[ha:HAS_ATTRIBUTE]-(a:Attribute {{name: $node_attribute}})-[hv:HAS_VALUE]-(av:AttributeValue)
+        MATCH (a)-[hs:HAS_SOURCE]-(pool:%(number_pool_kind)s)
+        WHERE
+            av.value >= $start_range and av.value <= $end_range
+            AND ({rel_filter("ha")})
+            AND ({rel_filter("hv")})
+            AND ({rel_filter("hs")})
+        """ % {
+            "node": self.pool.node.value,
+            "number_pool_kind": InfrahubKind.NUMBERPOOL,
+        }
+        self.add_to_query(query)
+
+        self.return_labels = ["n.uuid as id", "hv.branch as branch", "av.value as value"]
+        self.order_by = ["av.value"]
+
+
+class NumberPoolGetReserved(Query):
+    name: str = "numberpool_get_reserved"
+
+    def __init__(
+        self,
+        pool_id: str,
+        identifier: str,
+        **kwargs: dict[str, Any],
+    ) -> None:
+        self.pool_id = pool_id
+        self.identifier = identifier
+
+        super().__init__(**kwargs)  # type: ignore[arg-type]
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:
+        self.params["pool_id"] = self.pool_id
+        self.params["identifier"] = self.identifier
+
+        query = """
+        MATCH (pool:%(number_pool)s { uuid: $pool_id })-[rel:IS_RESERVED]->(reservation:AttributeValue)
+        WHERE rel.identifier = $identifier
```

Comment:
I think we are missing some filters in this query to. ensure that the relationship is active at this time

Reply:
Yes that makes sense. This part is pretty much copied from the address and prefix pool queries. I'll add it here to have it in place. But I think we also need to consider how to deallocate resources from our pools. If we should trigger actions if nodes / relationships are deleted and if we need something else to just remove an allocation.

Reply:
@dgarros, I pushed a change to add this, are you good to merge this now?

(will also open another issue for proper support to update the allocations themselves for all of the pools).